packages/overlay/src/overlay-stack.ts

```diff
@@ -585,8 +584,20 @@ export class OverlayStack {
         if (this.preventMouseRootClose || event.defaultPrevented) {
             return;
         }
-        this.closeTopOverlay();
         const overlaysToClose = [];
+        // Find the top most overlay that is not triggered by an
+        // element on the path of the current click event.
+        let index = this.overlays.length;
+        while (index && overlaysToClose.length === 0) {
+            index -= 1;
+            const overlay = this.overlays[index];
+            if (
+                !event.composedPath().includes(overlay.trigger) ||
+                overlay.interaction !== 'hover'
```

Comment:
Perfect. This should keep overlays open that are in the same path (i.e. an open modal that has a click overlay), and it should keep hover overlays open as well (they go away when you remove hover). NAILED IT!